Tribler/Test/Core/Modules/RestApi/test_util.py

```diff
@@ -1,35 +1,62 @@
-from Tribler.Core.Modules.restapi.util import convert_torrent_to_json
-from Tribler.Test.Core.base_test import TriblerCoreTest
+import struct
+
+from Tribler.Core.Modules.restapi.util import convert_search_torrent_to_json, convert_db_channel_to_json, \
+    relevance_score_remote_torrent
+from Tribler.Core.Session import Session
+from Tribler.Test.Core.base_test import TriblerCoreTest, MockObject
 
 
 class TestRestApiUtil(TriblerCoreTest):
     """
     This class contains various tests for the rest api utility methods.
     """
 
+    def tearDown(self, annotate=True):
+        TriblerCoreTest.tearDown(self, annotate=annotate)
+        Session.del_instance()  # We are opening a session when assigning a relevance score to a remote torrent
+
     def test_convert_torrent_to_json_dict(self):
         """
         Test whether the conversion from remote torrent dict to json works
         """
+        mocked_db = MockObject()
+        mocked_db.latest_matchinfo_torrent = None
+        Session.get_instance().open_dbhandler = lambda _: mocked_db
+
         input = {'torrent_id': 42, 'infohash': 'a', 'name': 'test torrent', 'length': 43,
                  'category': 'other', 'num_seeders': 1, 'num_leechers': 2}
         output = {'id': 42, 'infohash': 'a'.encode('hex'), 'name': 'test torrent', 'size': 43, 'category': 'other',
-                  'num_seeders': 1, 'num_leechers': 2, 'last_tracker_check': 0}
-        self.assertEqual(convert_torrent_to_json(input), output)
+                  'num_seeders': 1, 'num_leechers': 2, 'last_tracker_check': 0, 'relevance_score': 0.0}
```

Comment:
Here the inconsistency would show if you added a test where the score would be set to 0 but the list would be empty, returning 0 instead of 0.0